iggy/src/users/change_password.rs

```diff
@@ -0,0 +1,193 @@
+use crate::bytes_serializable::BytesSerializable;
+use crate::command::CommandPayload;
+use crate::error::Error;
+use crate::identifier::Identifier;
+use crate::users::{MAX_PASSWORD_LENGTH, MIN_PASSWORD_LENGTH};
+use crate::validatable::Validatable;
+use bytes::BufMut;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+use std::str::{from_utf8, FromStr};
+
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct ChangePassword {
+    #[serde(skip)]
+    pub user_id: Identifier,
+    pub current_password: String,
+    pub new_password: String,
+}
+
+impl CommandPayload for ChangePassword {}
+
+impl Default for ChangePassword {
+    fn default() -> Self {
+        ChangePassword {
+            user_id: Identifier::default(),
+            current_password: "secret".to_string(),
+            new_password: "topsecret".to_string(),
+        }
+    }
+}
+
+impl Validatable<Error> for ChangePassword {
+    fn validate(&self) -> Result<(), Error> {
+        if self.current_password.is_empty()
+            || self.current_password.len() > MAX_PASSWORD_LENGTH
+            || self.current_password.len() < MIN_PASSWORD_LENGTH
```

Comment:
I would skip this one - it is redundant - should be guaranteed by setting initial password

Reply:
It's just in case that someone would try to send an invalid password to be compared in the first place - the the command will immediately fail.